vcm_api/submission/test_submission_view.py

```diff
@@ -0,0 +1,154 @@
+import pytest
+import pytz
+from datetime import timedelta, datetime
+from vcm_api.contest.models import Contest
+from vcm_api.problem.models import Problem
+from vcm_api.user.models import Profile
+from django.contrib.auth import get_user_model
+from rest_framework.test import APIClient
+from rest_framework import status
+
+User = get_user_model()
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def test_user():
+    '''This fixture provides the test with user along with profile'''
+    test_user = User.objects.create(
+        first_name="test",
+        last_name="user",
+        username="gibberish",
+        password="test_me123")
+    return test_user
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def test_profile():
+    '''This fixture provides the test with a profile with valid online_judge usernames'''
+    test_profile = Profile.objects.create(
+        cf_handle="gtheoden42",
+        cc_handle="theoden42",
+        ac_handle="gtheoden42")
+    return test_profile
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def test_contest():
+    '''This fixture provides the test with test contest with valid problems'''
+    test_problem1 = Problem.objects.create(
+        name="PermuTree",
+        link="https://codeforces.com/problemset/problem/1856/E1",
+        score=100,
+        online_judge="codeforces")
+    test_problem2 = Problem.objects.create(
+        name="TwoPiles",
+        link="https://www.codechef.com/problems/SPLITMIN",
+        score=100,
+        online_judge="codechef")
+    test_problem3 = Problem.objects.create(
+        name="Odd or Even",
+        link="https://atcoder.jp/contests/abc313/tasks/abc313_d",
+        score=100,
+        online_judge="atcoder")
+
+    test_contest = Contest.objects.create(name="CodeRumble", start_date_time=datetime(
+        2023, 8, 9, 12, 0, 0, 0, tzinfo=pytz.UTC), duration=timedelta(hours=5, minutes=30))
+    test_contest.problems.add(test_problem1)
+    test_contest.problems.add(test_problem2)
+    test_contest.problems.add(test_problem3)
+    return test_contest
+
+
+@pytest.mark.django_db
+def test_problemStatus_unauthenticatedUser_returns_unauthorized():
+    client = APIClient()
+    uri = '/api/submissions/'
+    response = client.post(uri, {"problem_id": 1})
+    assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+
+@pytest.mark.django_db
+def test_problemStatus_invalidProblemId_returns_bad_request(test_user):
+    client = APIClient()
+    uri = '/api/submissions/'
+    client.force_authenticate(user=test_user)
+    response = client.post(uri, {"problem_id": '101001'})
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+
+@pytest.mark.django_db
+def test_ProblemStatus_authenticatedUserIsNotProblemOwner_returns_forbidden(
```

Comment:
i think we should check for contest participant / owner / admin instead of problem owner?

Reply:
done!